src/datatypes/src/vectors/helper.rs

```diff
@@ -258,9 +258,10 @@ impl Helper {
         Ok(match array.as_ref().data_type() {
             ArrowDataType::Null => Arc::new(NullVector::try_from_arrow_array(array)?),
             ArrowDataType::Boolean => Arc::new(BooleanVector::try_from_arrow_array(array)?),
-            ArrowDataType::LargeBinary => Arc::new(BinaryVector::try_from_arrow_array(array)?),
-            ArrowDataType::FixedSizeBinary(_) | ArrowDataType::Binary => {
-                let array = arrow::compute::cast(array.as_ref(), &ArrowDataType::LargeBinary)
+            ArrowDataType::LargeBinary
+            | ArrowDataType::FixedSizeBinary(_)
+            | ArrowDataType::Binary => {
+                let array = arrow::compute::cast(array.as_ref(), &ArrowDataType::Binary)
```

Comment:
We don't need to cast the array if the array type is Binary.
```suggestion
            ArrowDataType::Binary => Arc::new(BinaryVector::try_from_arrow_array(array)?),
            ArrowDataType::FixedSizeBinary(_)
            | ArrowDataType::LargeBinary => {
                let array = arrow::compute::cast(array.as_ref(), &ArrowDataType::Binary)
```